lawnchair/src/app/lawnchair/smartspace/IcuDateTextView.kt

```diff
@@ -39,27 +51,62 @@ class IcuDateTextView @JvmOverloads constructor(
 
     private fun onTimeChanged(updateFormatter: Boolean) {
         if (isShown) {
-            if (formatter == null || updateFormatter) {
-                val formatter: DateFormat = DateFormat.getInstanceForSkeleton(
-                    context.getString(R.string.smartspace_icu_date_pattern), Locale.getDefault()
-                )
-                this.formatter = formatter
-                formatter.setContext(DisplayContext.CAPITALIZATION_FOR_BEGINNING_OF_SENTENCE)
+            val format = when (calendar) {
+                SmartspaceCalendar.Persian -> getTimeTextPersian(updateFormatter = updateFormatter)
+                else -> getTimeTextGregorian(updateFormatter = updateFormatter)
             }
-            val format = formatter!!.format(java.lang.Long.valueOf(System.currentTimeMillis()))
             if (text != format) {
+                textAlignment =
+                    if (calendar == SmartspaceCalendar.Persian) TEXT_ALIGNMENT_TEXT_END else TEXT_ALIGNMENT_TEXT_START
                 text = format
                 contentDescription = format
             }
         }
     }
 
+    private fun getTimeTextPersian(updateFormatter: Boolean): String {
+        if (formatterPersian == null || updateFormatter) {
+            val formatter = PersianDateFormat(
+                context.getString(R.string.smartspace_icu_date_pattern_persian),
+                PersianDateFormat.PersianDateNumberCharacter.FARSI
+            )
+            this.formatterPersian = formatter
+        }
+        return formatterPersian!!.format(PersianDate(java.lang.Long.valueOf(System.currentTimeMillis())))
+    }
```

Comment:
I’d suggest avoiding the use of not-null assertion here.

```suggestion
    private fun getTimeTextPersian(updateFormatter: Boolean): String {
        val formatter = formatterPersian.takeIf { updateFormatter.not() } ?: PersianDateFormat(
            context.getString(R.string.smartspace_icu_date_pattern_persian),
            PersianDateFormat.PersianDateNumberCharacter.FARSI,
        ).also { formatterPersian = it }
        return formatter.format(PersianDate(java.lang.Long.valueOf(System.currentTimeMillis())))
    }
```